go/host/db/batches.go

```diff
@@ -0,0 +1,112 @@
+package db
```

Comment:
This should have a unit test file to accompany it. It's nicely written and decoupled should be easy to inject mock objects and test the batches logic